src/Compilers/VisualBasic/Portable/Analysis/FlowAnalysis/ReadWriteWalker.vb

```diff
@@ -83,18 +92,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End If
         End Sub
 
+        ' range variables are only returned in the captured set if inside the region
         Private Sub NoteCaptured(variable As Symbol)
-            If variable.Kind <> SymbolKind.RangeVariable Then
+            If _regionPlace = RegionPlace.Inside Then
+                _capturedInside.Add(variable)
+                _captured.Add(variable)
+            ElseIf variable.Kind <> SymbolKind.RangeVariable Then
+                _capturedOutside.Add(variable)
```

Comment:
Can we reach this? It looks like this was previously throwing UnexpectedValue

Reply:
There is an empty case clause for RegionPlace.Before and RegionPlace.After.